src/Aspire.Hosting/ApplicationModel/InputAnnotation.cs

```diff
@@ -49,11 +45,25 @@ public InputAnnotation(string name, string? type = null, bool secret = false)
     /// </summary>
     public InputDefault? Default { get; set; }
 
-    /// <summary>
-    /// The value of the input.
-    /// </summary>
-    public string GenerateDefaultValue()
+    internal Func<string>? ValueGetter { get; set; }
```

Comment:
This `ValueGetter` is the only piece I don't really like about this change. But I'm not sure how to make it better. Curious to hear others thoughts.

One thought I had was to make a new class that derives from `InputDefault`, but that felt like over-engineering. And it would need to be filtered out during manifest generation, which didn't feel great.

Another concern I had is that since ParameterResource's value is a callback, it can change from invocation to invocation, so I didn't let the value "cache" in that case. Would it be better to only invoke the callback the first time?

Reply:
ParameterResource's value should be read once and stored as well.

Reply:
Ok, this is now done. I feel a little better about this code now.